internal/configs/transportserver.go

```diff
@@ -154,7 +154,7 @@ func generateTransportServerHealthCheck(upstreamName string, generatedUpstreamNa
 			}
 
 			if u.HealthCheck.Port > 0 {
-				hc.Port = u.HealthCheck.Port
+				hc.Port = &u.HealthCheck.Port
```

Comment:
Assigning a pointer from a loop variable can be a source of subtle bugs.  Please see an example here https://go.dev/play/p/5pKPotijvmX 

In this case, because `HealthCheck` itself is a pointer, it doesn't look like we have a bug here. However, it is not clear from the code, and as a result, it can make the reader confused.

Considering that, I wonder if it makes sense to get rid of the pointer all together - a use `int` both in `StreamHealthCheck` in `stream.go`  and `HealthCheck` in `http.go`.

For that case, the statement `{{ if $s.HealthCheck.Port }} port={{ $s.HealthCheck.Port }}` will never evaluate if the value of `HealthCheck.Port` is `0`. So the existing logic would still work. This applies to the VirtualServer too. 

Also, we will be able to simplify conditions like below:
```
if u.HealthCheck.Port > 0 {
	hc.Port = u.HealthCheck.Port
	hc.Port = &u.HealthCheck.Port
```
->
```
hc.Port = u.HealthCheck.Port 
```
and
```
	if upstream.HealthCheck.Port > 0 {
		hc.Port = &upstream.HealthCheck.Port
	}
```
->
```
hc.Port = upstream.HealthCheck.Port
```

I apologies - my comment here https://github.com/nginxinc/kubernetes-ingress/pull/2378#discussion_r792134992 was a suggestion to convert int to a pointer for stream (to be consistent with http). However, now that it is implemented, I noticed a potential problem with that  :(

Reply:
Ah, good catch! That makes sense - thanks Michael! I'll update this 👍🏼